Fail to build on Fedora 35 rawhide

Fedora 35 is under development, so it might be a bug in the setup, or it can be an early warning that will happen in a newer toolchain. Builds on Fedora 34 and Fedora 33.
```
g++ -c -pipe -std=c++1z -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -DNDEBUG -fomit-frame-pointer -O2 -std=gnu++1z -w -D_REENTRANT -fPIC -DQT_DEPRECATED_WARNINGS -DUSE_QT_IN_BITCOIN -DHAVE_ENDIAN_H -DHAVE_DECL_HTOBE16 -DHAVE_DECL_HTOLE16 -DHAVE_DECL_BE16TOH -DHAVE_DECL_LE16TOH -DHAVE_DECL_HTOBE32 -DHAVE_DECL_HTOLE32 -DHAVE_DECL_BE32TOH -DHAVE_DECL_LE32TOH -DHAVE_DECL_HTOBE64 -DHAVE_DECL_HTOLE64 -DHAVE_DECL_BE64TOH -DHAVE_DECL_LE64TOH -DHAVE_DECL___BUILTIN_CLZL -DHAVE_DECL___BUILTIN_CLZLL -DGIT_COMMIT=\"\" -DENABLE_ZMQ -DHAVE_JEMALLOC_HEADERS -DQT_NO_DEBUG -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -Isrc -I/usr/include/qt5 -I/usr/include/qt5/QtNetwork -I/usr/include/qt5/QtCore -I. -I/usr/lib64/qt5/mkspecs/linux-g++ -o Util.o src/Util.cpp
/bin/sh: line 1: git: command not found
In file included from /usr/include/pthread.h:33,
                 from /usr/include/c++/11/x86_64-redhat-linux/bits/gthr-default.h:35,
                 from /usr/include/c++/11/x86_64-redhat-linux/bits/gthr.h:148,
                 from /usr/include/c++/11/ext/atomicity.h:35,
                 from /usr/include/c++/11/bits/basic_string.h:39,
                 from /usr/include/c++/11/string:55,
                 from /usr/include/qt5/QtCore/qbytearray.h:52,
                 from /usr/include/qt5/QtCore/QByteArray:1,
                 from src/ByteView.h:21,
                 from src/Util.h:21,
                 from src/Mixins.h:21,
                 from src/App.h:21,
                 from src/Util.cpp:19:
src/Util.cpp:66:48: error: call to non-'constexpr' function 'long int __sysconf(int)'
   66 | static constexpr unsigned PLATFORM_STACK_MIN = PTHREAD_STACK_MIN;
      |                                                ^~~~~~~~~~~~~~~~~
/usr/include/bits/pthread_stack_min-dynamic.h:24:17: note: 'long int __sysconf(int)' declared here
   24 | extern long int __sysconf (int __name) __THROW;
      |                 ^~~~~~~~~
make: *** [Makefile:1896: Util.o] Error 1
```

gcc-c++: 11.2.1
qt5-qtbase-devel: 5.15.2
glibc-devel: 2.34

Full build log: https://download.copr.fedorainfracloud.org/results/jonny/BitcoinCash/fedora-rawhide-x86_64/02404254-fulcrum/build.log.gz

--- a/src/Util.cpp
+++ b/src/Util.cpp
@@ -60,67 +60,7 @@
 #include <mutex>
 #include <thread>
 
-#if __has_include(<pthread.h>) && !defined(Q_OS_WIN)
-// MacOS, Linux, etc
-#  include <pthread.h>
-static constexpr unsigned PLATFORM_STACK_MIN = PTHREAD_STACK_MIN;
-#elif defined(Q_OS_WIN) && __has_include(<sysinfoapi.h>)
-// Windows
-#  include <sysinfoapi.h>
-// typically 4KiB to 64KiB
-static const unsigned PLATFORM_STACK_MIN = []{
-    SYSTEM_INFO si;
-    GetSystemInfo(&si);
-    return unsigned(si.dwPageSize);
-}();
-#else
-// unknown platform, 0 indicates to use defaults
-static constexpr unsigned PLATFORM_STACK_MIN = 0;
-#endif
-
 namespace Util {
-    unsigned getPlatformMinimumThreadStackSize() {
-        // we must probe the minimum stack size just in case PTHREAD_STACK_MIN is a lie (it can happen if we built the
-        // static binary for a slightly different platform/compiler than what it is executing on now).
-        static const unsigned probedMin = [] {
-            auto supp = App::addQtLogSuppression("QThread::start"); // suppress the error Qt may generate while we probe
-            Defer d([&supp]{ App::rmQtLogSuppression(supp);}); // undo the suppression on scope end
-            try {
-                if (!PLATFORM_STACK_MIN)
-                    return 0U;
-                const auto t0 = getTimeMicros();
-                std::promise<unsigned> p;
-                auto fut = p.get_future();
-                std::unique_ptr<QThread> thr {QThread::create([p=std::move(p)]() mutable {
-                    p.set_value(QThread::currentThread()->stackSize());
-                })};
-                thr->setStackSize(PLATFORM_STACK_MIN);
-                thr->setObjectName("ProbeMinStackSize");
-                thr->start();
-                Defer deferredJoin([&thr]{
-                    if (thr && !thr->isFinished()) {
-                        thr->terminate();
-                        if (!thr->wait(250))
-                            // sadly, we must release the unique_ptr since deleting the thread while it's running can cause a crash.
-                            Error() << "Timed-out waiting for thread \"" << thr.release()->objectName() << "\". FIXME!";
-                    }
-                });
-                if (auto res = fut.wait_for(std::chrono::milliseconds(250)); res == std::future_status::ready) {
-                    const auto ret = fut.get();
-                    Debug() << "successfully probed minimum stack size of " << ret << " in "
-                            << (getTimeMicros()-t0) << " usec";
-                    return ret;
-                } else
-                    Warning() << "Failed to probe thread minimum stack size, using system default.";
-            } catch (const std::exception &e) {
-                // this should never be reached
-                Error() << "Exception caught in minimum stack size probe function: " << e.what();
-            }
-            return 0U;
-        }();
-        return probedMin;
-    }
-
     QString basename(const QString &s) {
         QRegExp re("[\\/]");
         auto toks = s.split(re);

--- a/src/Util.h
+++ b/src/Util.h
@@ -932,11 +932,6 @@ namespace Util {
         };
     } // end namespace AsyncSignalSafe
 
-    /// Returns the minimum stack size in bytes for this platform, suitable for passing to QThread::setStackSize.
-    /// If the platform is unknown to this codebase, then 0 is returned (which if passed to QThread::setStackSize
-    /// just results in using the default thread stack settings).
-    unsigned getPlatformMinimumThreadStackSize();
-
     struct MaxOpenFilesResult {
         enum Status {
             Ok = 0, NotRelevant, Error

